enterprise/ha/src/main/java/org/neo4j/kernel/ha/cluster/SwitchToSlave.java

```diff
@@ -432,8 +447,19 @@ private void catchUpWithMaster() throws IllegalArgumentException, InterruptedExc
         userLog.info( "Catching up with master. I'm at %s", catchUpRequestContext );
 
         // Unpause the update puller, because we know that we are a slave that just started communication with master.
-        updatePuller.unpause();
-        updatePuller.await( UpdatePuller.NEXT_TICKET, true );
+        while ( true )
```

Comment:
It is guaranteed to terminate. Unpause will always work within a very short while.